perfkitbenchmarker/resource.py

```diff
@@ -142,15 +143,39 @@ def _DeleteResource(self):
     if not self.delete_end_time:
       self.delete_end_time = time.time()
 
-  def Create(self):
-    """Creates a resource and its dependencies."""
+  def Create(self, ready_poll_interval=vm_util.POLL_INTERVAL,
+             ready_max_retries=vm_util.MAX_RETRIES, ready_timeout=None,
+             ready_fuzz=vm_util.FUZZ,
+             ready_log_errors=True):
+    """Creates a resource and its dependencies.
+
+    Args:
+      ready_poll_interval: Time between _IsReady checks.  Only
+          useful if the subclass implements _IsReady.
+      ready_max_retires: Maximum number of _IsReady retries.
+      ready_timeout: Maximum time to wait for the object to be
+          ready.
+      ready_fuzz: The fuzz on the _IsReady checks.  Only useful
+          if the subclass implements _IsReady.  See vm_util.Retry
+          for details.
+      ready_log_errors: A boolean describing whether errors should
+          be logged for the _IsReady check.
+    """
+    @vm_util.Retry(poll_interval=ready_poll_interval,
```

Comment:
I think accepting optional retry parameters to Create is probably a good thing, but only applying them to the "waituntilready" and not "createresource" itself seems like it might create confusion to me. Also if we are going to allow this, we may also want these options available to Delete as well